src/main/scala/mesosphere/marathon/api/CORSFilter.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.marathon.api
+
+import javax.inject.{ Inject }
+import javax.servlet._
+import javax.servlet.http.{ HttpServletRequest, HttpServletResponse }
+import mesosphere.marathon.{ MarathonConf }
+
+import scala.collection.JavaConverters._
+
+class CORSFilter @Inject() (config: MarathonConf) extends Filter {
+  override def init(filterConfig: FilterConfig): Unit = {}
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+
+    response match {
+      case httpResponse: HttpServletResponse if config.accessControlAllowOrigin.isSupplied =>
+        val httpRequest = request.asInstanceOf[HttpServletRequest]
+
+        // Map access_control_allow_origin flag into separate headers
+        val origins: Seq[String] = config.accessControlAllowOrigin()
```

Comment:
Could you move this out to a lazy val outside the scope of this method to avoid recomputation?